### ISSUE TITLE
Property 'menu' does not exist on type 'IntrinsicAttributes & DropdownProps'.

### ISSUE THREAD

**Issue by hnbzlb** [ant-design/ant-design#38344]
### Reproduction link

[https://ant.design/components/dropdown/](https://ant.design/components/dropdown/)

### Steps to reproduce

```javascript
const items: MenuProps['items'] = [
    {
      label: '1st menu item',
      key: '1',
    },
    {
      label: '2nd menu item',
      key: '2',
    },
    {
      label: '3rd menu item',
      key: '3',
    },
  ]; 
<Dropdown  menu={items}>
        <Button>
          <DownOutlined />
        </Button>
      </Dropdown>
```

### What is expected?

VScode compiles successfully.

### What is actually happening?

```javascript
TS2322: Type '{ children: Element; menu: ItemType[]; }' is not assignable to type 'IntrinsicAttributes & DropdownProps'.
  Property 'menu' does not exist on type 'IntrinsicAttributes & DropdownProps'.
    115 |         items={items}
    116 |       />
  > 117 |       <Dropdown  menu={items}>
        |                  ^^^^
    118 |         <Button>
    119 |           <DownOutlined />
    120 |         </Button>
```

| Environment | Info |
| --- | --- |
| antd | 4.23.6 |
| React | 18.2.0 |
| System | window 10，nodejs: 16.17.1 |
| Browser | chrome |

**Comment by kiner-tang** [comment]
you should pass 'items' like this

![image](https://user-images.githubusercontent.com/10286961/199455429-8c06a4fa-27ba-4010-9360-0b58006d12da.png)

**Comment by MadCcc** [comment]
menu is available after 4.24.0

**Comment by MadCcc** [comment]
menu is available after 4.24.0
